### PR TITLE
Change high frequency free running timers to on-demand tasks

### PR DESCRIPTION
**Pull request by omerfirmak**
We could maybealso increase their firing delay from 10 msecs to something more reasonable

### CODE REVIEW

**Comment by Geod24** [comment]
I think you can reuse the same instance, and call `rearm` on it, instead of calling `setTimer` again.

**Comment by Geod24** [comment]
Also if it's not period, do you need to `stop` it ?

**Comment by omerfirmak** [comment]
> 
> 
> I think you can reuse the same instance, and call `rearm` on it, instead of calling `setTimer` again.

Our common interface only has stop, that might be a vibe.d thing maybe.

https://github.com/bosagora/agora/blob/236b2ee0d98254c498bef5eda9ef5539fd785b10/source/agora/common/Task.d#L96-L105

**Comment by omerfirmak** [comment]
> 
> 
> Also if it's not period, do you need to `stop` it ?

If it is enabled (non-null timer), we dont want it to fire one last time after `shutdown`. This might not work as expected with localrest tho, I'll check.

**Comment by omerfirmak** [comment]
Ok, it works fine in LocalRest too

https://github.com/Geod24/localrest/blob/3b650c53ea8af30061a7feab5af43992ad88655b/source/geod24/LocalRest.d#L537-L538

**Comment by Geod24** [comment]
> Our common interface only has stop

Well, you know what to do :D

**Comment by Geod24** [comment]
> If it is enabled (non-null timer), we dont want it to fire one last time after `shutdown`.

I agree, but that would only happen if we re-enter the event loop.

**Comment by omerfirmak** [comment]
> 
> 
> > If it is enabled (non-null timer), we dont want it to fire one last time after `shutdown`.
> 
> I agree, but that would only happen if we re-enter the event loop.

Then why call it for the periodic one either? I think it's better being safe than sorry :D

File: source/agora/network/Client.d [hunk 1]
```diff
@@ -168,7 +171,9 @@ public class NetworkClient
         this.exception = new Exception(
             format("Request failure to %s after %s attempts", address,
                 max_retries));
-        this.gossip_timer = this.taskman.setTimer(10.msecs, &this.gossipTask, Periodic.Yes);
+        // Create and stop timer immediately
+        this.gossip_timer = this.taskman.setTimer(GossipDelay, &this.gossipTask, Periodic.No);
+        this.gossip_timer.stop();
```
Comment (Geod24) [inline source/agora/network/Client.d:176]: https://github.com/vibe-d/vibe-core/blob/9ca874dc75020b1488a0ac64a1cc5f6471ba15f4/source/vibe/core/core.d#L1243-L1255

But okay, this works :D

File: source/agora/consensus/protocol/Nominator.d [hunk 2]
```diff
@@ -189,12 +192,14 @@ extern(D):
         this.ledger = ledger;
         this.enroll_man = enroll_man;
         this.scp_envelope_store = new SCPEnvelopeStore(cacheDB);
+        // Create and stop timer immediately
+        this.envelope_timer = this.taskman.setTimer(10.msecs,
```
Comment (Geod24) [inline source/agora/consensus/protocol/Nominator.d]: `EnvTaskDelay` ?

**Review by Geod24 — APPROVED** [review]
LGTM. The `rearm` thing is dangerous, because adding a new method can go wrong very easily, but luckily this doesn't change much.

**Comment by Geod24** [comment]
Did the minor change directly, merging

**Pull request merged**